Prevent hyphenation of urls starting with https and E-Mail addresses

HKandulla opened the pull request:
Hi,
currently only URLs starting with `http:` or `www.` are prevented from hyphenation. I have added `https:` to the regex to include URLs starting with https.
Best, Hannes

finwe commented:
Please, add unit tests and a changelog entry. Then rebase onto the development branch which fixes float errors in tests.

finwe commented:
Also, remove the `fix:` prefix from the commit message (as this is not entirely a fix) and start the message with a capital leter.

HKandulla commented:
Hi @finwe,
thanks for your quick response. I added unit tests and changelog entry and rebased onto development branch.
I also included [https://github.com/mpdf/mpdf/pull/1634](url) in this PR.
If there is anything else, let me know.
Best, Hannes

finwe commented:
Cool, thanks!

HKandulla commented:
Hi @finwe ,
I just realised that due to this part: [https://github.com/mpdf/mpdf/blob/cc1759f153c948c76b0ef54a39a6186be9965def/src/Mpdf.php#L8136](https://github.com/mpdf/mpdf/blob/cc1759f153c948c76b0ef54a39a6186be9965def/src/Mpdf.php#L8136)
URLs and E-Mail addresses still behave differently.

E-Mail addresses would break before Hyphen-minus.
URLs would not break at all

Do you think this behaviour is desired? I would suggest E-Mail addresses and urls should break before Hyphen-minus.
Please let me know whether I should prepare such a PR including tests, etc.
Thanks, Hannes
